### ISSUE TITLE
Problème à l'installation

### ISSUE THREAD

**Issue by 15bart15** [broncowdd/BoZoN#101]
J'utilise Bozon depuis quelque temps déjà, et là j'ai voulu mettre la dernière version. Je supprime l'ancien dossier (version 1.xxx quelque chose), je dézippe le nouveau. Je crée un compte, et lors du login: 'You are definitely NOT from here !'. Euh qu'est ce qu'il me dit?? Dans le log de nginx , j'ai le message suivant: 'Undefined variable: salt in bozon/core/auto_restrict.php on line 96'.
J'ai bien php5-gd et ziparchive d'installé. Je ne comprends pas.

**Comment by fpp-gh** [comment]
Je venais pour ouvrir le même, merci de l'avoir fait pour moi :-)

Petite différence au départ : j'ai installé cette version hier "from scratch", sans rien récupérer de l'ancienne.

Ça a bien marché jusqu'à cet après-midi, et là, boum :-)

**Comment by 15bart15** [comment]
> Petite différence au départ : j'ai installé cette version hier "from scratch", sans rien récupérer de l'ancienne.

Je me suis mal exprimé peut-être, j'ai tout supprimé aussi.

**Comment by fpp-gh** [comment]
Dans mon cas je soupçonne aussi les mesures prises par auto_restrict.php pour protéger le login de l'utilisateur admin, qui semble vouloir une URL d'accès immuable.

Mon nom de domaine résout à la fois domain.ext et www.domain.ext vers la machine d'hébergement.

Hier j'ai réinstallé Bozon en utilisant l'adresse en www dans le navigateur, et aujourd'hui j'ai dû y accéder par hasard avec l'adresse sans www, et c'est là qu'il a perdu les pédales (wrong token, not from here, IP ban et tout le toutim :-).

En revanche je n'ai pas pu touver de référence en clair à ce nom de domaine dans les divers fichiers, donc j'ai re-écrasé/installé en attendant l'explication ...

**Comment by broncowdd** [comment]
Mmmm... Intéressant. Je vais me pencher dessus et essayer de reproduire le problème sur mon serveur. Merci :-)

Le 16 mars 2016 22:05:29 CET, fpp-gh <notifications@github.com> a écrit :
>Dans mon cas je soupçonne aussi les mesures prises par
>auto_restrict.php pour protéger le login de l'utilisateur admin, qui
>semble vouloir une URL d'accès immuable.
>
>Mon nom de domaine résout à la fois domain.ext et www.domain.ext vers
>la machine d'hébergement.
>
>Hier j'ai réinstallé Bozon en utilisant l'adresse en www dans le
>navigateur, et aujourd'hui j'ai dû y accéder par hasard avec l'adresse
>sans www, et c'est là qu'il a perdu les pédales (wrong token, not from
>here, IP ban et tout le toutim :-).
>
>En revanche je n'ai pas pu touver de référence en clair à ce nom de
>domaine dans les divers fichiers, donc j'ai re-écrasé/installé en
>attendant l'explication ...
>
>
>---
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/broncowdd/BoZoN/issues/101#issuecomment-197549177

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by broncowdd** [comment]
Bon, après réflexion, je pense que ça vient du fait que tu as mis à jour une version 1.7 ou inférieure, dans laquelle ma lib de gestion des utilisateurs ne gérait pas encore le multiuser. 
La façon la plus simple de régler le problème est de supprimer les fichiers private/auto_restrict_users.php, auto_restrict_data.php. 
Normalement, Bozon recréera ses fichiers quand tu recréeras ton compte.

Pour le problème de sous domaine, je regarde aujourd'hui.

**Comment by fpp-gh** [comment]
Euh, pas du tout :-)
Comme indiqué dans le 2e message, je n'ai rien mis à jour mais installé le "master" 2.3 ex nihilo...

D'ailleurs il m'a refait le coup du "not from here" ce matin alors que cette fois je suis sûr qu'il n'y a pas eu de mélange d'URL... j'ai pu ratrapper en récupérant la session en cours dans l'historique, mais j'aimerais bien comprendre ce qui le perturbe comme ça, histoire d'éviter...

Quand on supprime les auto-restrict comme ci-dessus, et qu'on recrée le user à l'identique, on retrouve ses fichiers ou pas ? :-)

**Comment by 15bart15** [comment]
Pareil, installation toute neuve, aucune récupération des anciennes données. Rien
Et contrairement à fpp-gh je n'ai accès à rien du tout.
Peut-être nginx qui fout la zone? 
@broncowdd J'ai lu quelque part que tu ne l'utilisais pas.
@fpp-gh tu utilises quoi?

**Comment by fpp-gh** [comment]
nginx aussi, forcé en https avec certificat Let's Encrypt

Que des premières en auto-hébergement, tout ça c'est de la science-fiction pour moi, mais ça semble bien marcher avec tout le reste :-)

**Comment by broncowdd** [comment]
Oui, il faudra se pencher sérieusement sur https... J'ai pas eu le temps mais ce sera indispensable. 

Le 17 mars 2016 10:55:36 CET, fpp-gh <notifications@github.com> a écrit :
>nginx aussi, forcé en https avec certificat Let's Encrypt
>
>Que des premières en auto-hébergement, tout ça c'est de la
>science-fiction pour moi, mais ça semble bien marcher avec tout le
>reste :-)
>
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/broncowdd/BoZoN/issues/101#issuecomment-197795569

-- 
Envoyé de mon appareil Android avec K-9 Mail. Veuillez excuser ma brièveté.

**Comment by 15bart15** [comment]
Pas de HTTPS pour moi.
Serveur hébergé sur un RaspberryPi uniquement pour un intranet familial.

**Comment by fpp-gh** [comment]
Tiens, peut-être un moyen simple de reproduire le problème pour analyse :
1) je suis loggué sur mon BoZon dans un des what mille onglets de mon Firefox
2) ayant oublié, j'en crée un nouveau, avec le marque-page /index.php&p=login
3) paf, not from here !
4) ah oui zut chuis déjà loggué, donc si je garde que index.php ça doit suffire ?
5) paf, bannedfuckoff ! :-)
6) je retrouve l'onglet oublié, pareil...
7) mais curieusement, après quelques "back" ça revient...
Bref on vit dangereusement ! :-)

**Comment by broncowdd** [comment]
Je m'y attelle ^^

**Comment by broncowdd** [comment]
au passage, on ne peut se rendre directement vers la page login car auto_restrict bloque les post et get sans token, pour éviter des manipulations non autorisées.
